### PR TITLE
Windows audio (WIP)

### PR DESCRIPTION
**Pull request by calcitem**
Support audio on the Windows platform.

Based on https://github.com/calcitem/Sanmill/pull/421

Know issue:
Stoping sound before playing a new sound has not been implemented, we need to wait for AI Vs. AI is stable and then continues to debug it.

### CODE REVIEW

**Review by Leptopoda — CHANGES_REQUESTED** [review]

File: src/ui/flutter_app/lib/services/audios.dart [hunk 1]
```diff
@@ -154,32 +211,33 @@ class Audios {
...
+      return;
+    }
 
-      // If the platform is Windows [_initialized] should be false thus this code shouldn't be executed
-      if (Platform.isWindows) {
-        assert(false);
-      }
+    if (Platform.isWindows) {
+      await _playSound(sound);
+      return;
```
Comment (Leptopoda) [inline src/ui/flutter_app/lib/services/audios.dart:236]: oh and you should also directly return the future.

File: src/ui/flutter_app/lib/services/audios.dart [hunk 2]
```diff
@@ -60,12 +61,12 @@ class Audios {
...
   static Future<void> loadSounds() async {
     if (Platform.isWindows) {
-      debugPrint("$_tag Audio Player does not support Windows.");
+      debugPrint("$_tag Sound pool does not support Windows.");
       return;
     }
 
     if (_initialized) {
-      debugPrint("$_tag Audio Player is already initialized.");
+      debugPrint("$_tag Sound pool is already initialized.");
```
Comment (Leptopoda) [inline src/ui/flutter_app/lib/services/audios.dart:69]: also why did you change the logs? The thing already being initialized is sound pool not audio player. please change them back

**Review by Leptopoda — CHANGES_REQUESTED** [review]

**Comment by Leptopoda** [comment]
Also please revert your last commit on dev (the about_page refractor) and open a PR for it.
I have a few ideas on how to improve the code (things that don't need to be now are nullable ...) and having it in a PR would allow us to discuss those changes.

File: src/ui/flutter_app/lib/screens/about_page.dart [hunk 3]
```diff
@@ -136,10 +136,18 @@ class AboutPage extends StatelessWidget {
     await launch(url);
   }
 
-  Future<void> _launchFeedback() async {
+  Future<String?> _getCurrentLocale() async {
```
Comment (Leptopoda) [inline src/ui/flutter_app/lib/screens/about_page.dart:139]: This should be converted into a getter and maybe renamed to locale
``` dart
Future<String?> get locale asnyc {
  //some stuff
}
```

Also is there a reason not to use `Localizations.localeOf(context);` the result isn't nullable so it would make things a bit easier. We also wouldn't need to await the future. 

This would actually remove the need for the helper method all together

**Review by Leptopoda — CHANGES_REQUESTED** [review]

File: src/ui/flutter_app/lib/screens/about_page.dart (hunk 3 above)
Comment (calcitem) [reply src/ui/flutter_app/lib/screens/about_page.dart:139]: So what is the recommended way?  I didn't understand too much.

File: src/ui/flutter_app/lib/services/audios.dart [hunk 4]
```diff
@@ -112,11 +113,67 @@ class Audios {
...
+
   static Future<void> _playSound(Sound sound) async {
-    assert(!Platform.isWindows);
+    if (Platform.isWindows) {
+      return _playSoundForWindows(sound);
+    }
 
     final int soundId;
 
+    // TODO: Use map
```
[the remark this answers is not in the document]
Comment (calcitem) [reply src/ui/flutter_app/lib/services/audios.dart:176]: Do you mean？

[the remark this answers is not in the document]
Comment (Leptopoda) [reply src/ui/flutter_app/lib/services/audios.dart:176]: Switch case is better nullaware and when compiled more performant than a map

File: src/ui/flutter_app/lib/screens/about_page.dart (hunk 3 above)
Comment (Leptopoda) [reply src/ui/flutter_app/lib/screens/about_page.dart:139]: I think using `Localizations.localeOf(context)` is the prefered way.

**Comment by calcitem** [comment]
If there is user feedback that Windows cannot make a sound, we can ask him to download this PR.
